### PR TITLE
Issue 191: Run Python binding tests against Pravega standalone as part of build.

### LINKED ISSUE #191: Ensure Python integration tests are executed as part of build
**Problem description**
Ensure Python integration tests are executed as part of build

**Problem location**
build

**Suggestions for an improvement**
Ensure Python integration tests are executed as part of build

### PR DESCRIPTION
**Pull request by shrids**
**Change log description**  
Enable Python binding tests against Pravega standalone.

**Purpose of the change**  
Fixes #191 

**What the code does**  
Enable Python binding tests against Pravega standalone.

**How to verify it**  
All the existing tests and python binding tests should pass.

### CODE REVIEW

File: bindings/tox.ini [hunk 1]
```diff
@@ -9,7 +9,8 @@ requires = tox-pyo3
 pyo3 = True
 deps =
     pytest
-commands =
-    pytest -vvvv
+    pytest-timeout
+    aiounittest
+commands = pytest {posargs:tests} --timeout=300 -vvvvv
```
Comment (tkaitchuck) [inline bindings/tox.ini:14]: 5 v's?

**Review by tkaitchuck — APPROVED** [review]

File: bindings/tests/pravega_reader_test.py [hunk 2]
```diff
@@ -92,18 +96,18 @@ def test_multipleReader(self):
...
             self.assertEqual(b'data', event.data(), "Invalid event data")
 
         print("Number of events read after consuming slice2 ", count)
         self.assertEqual(count, 100, "100 events are expected")
 
-    # This test verifies data reading a Pravega stream with multiple readers.
-    # It also invokes release Segment after consuming part of the first segment slice and marks the first
-    # reader as offline. This test verifies if we are able to read all the 100 elements.
-    def test_multipleReaderPartialRead(self):
+    # # This test verifies data reading a Pravega stream with multiple readers.
```
Comment (Tristan1900) [inline bindings/tests/pravega_reader_test.py:107]: I guess the `#` is duplicate here and the next two lines?

**Review by Tristan1900 — APPROVED** [review]

File: bindings/tests/pravega_reader_test.py (hunk 2 above)
Comment (shrids) [reply bindings/tests/pravega_reader_test.py]: fixed it.

File: bindings/tox.ini (hunk 1 above)
Comment (shrids) [reply bindings/tox.ini:14]: this increases the verbosity level of the pytest.

**Comment by shrids** [comment]
The reason why this is timing out is because pravega-standalone does not start correctly inside the GitHub runner. Simulated this issue using https://github.com/nektos/act and realized that the Standalone never comes up on the GitHub runner.

File: .github/workflows/python_test.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '11' # The JDK version to make available on the path.
+      - name: Download and Run Pravega standalone
+        run: |
+          wget https://github.com/Tristan1900/pravega/releases/download/hostname/pravega-0.9.0-2711.b899fe6-SNAPSHOT.tgz
+          tar -xzvf pravega-0.9.0-2711.b899fe6-SNAPSHOT.tgz
+          pravega-0.9.0-2711.b899fe6-SNAPSHOT/bin/pravega-standalone > pravega.log 2>&1 &
+          sleep 120 && echo "Started standalone"
```
Comment (tkaitchuck) [inline .github/workflows/python_test.yml:30]: It would be good if we could identify a log line when it is ready and wait for that to be logged.

**Review by tkaitchuck — APPROVED** [review]

**Pull request merged**